src/libvirtApi/domain.js

```diff
@@ -685,20 +683,20 @@ export async function domainGet({
         else
             console.warn(`GET_VM failed for ${objPath}, undefining: ${ex.toString()}`);
         // but undefine either way -- if we  can't get info about the VM, don't show it
-        await store.dispatch(undefineVm({ connectionName, id: objPath }));
+        store.dispatch(undefineVm({ connectionName, id: objPath }));
     }
 }
 
-export function domainGetAll({ connectionName }) {
-    return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'ListDomains', [0], { timeout, type: 'u' })
-            .then(objPaths => {
-                store.dispatch(deleteUnlistedVMs(connectionName, [], objPaths[0]));
-                return Promise.all(objPaths[0].map(path => domainGet({ connectionName, id: path })));
-            })
-            .catch(ex => {
-                console.warn('GET_ALL_VMS action failed:', ex.toString());
-                return Promise.reject(ex);
-            });
+export async function domainGetAll({ connectionName }) {
+    try {
+        const [objPaths] = await call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'ListDomains', [0],
+                                      { timeout, type: 'u' });
+        store.dispatch(deleteUnlistedVMs(connectionName, [], objPaths));
+        await Promise.all(objPaths.map(path => domainGet({ connectionName, id: path })));
+    } catch (ex) {
+        console.warn('GET_ALL_VMS action failed:', ex.toString());
+        throw ex;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1280-20240125-153407-61755b5b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)